aio/content/examples/form-validation/src/app/shared/forbidden-name.directive.ts

```diff
@@ -22,7 +22,7 @@ export function forbiddenNameValidator(nameRe: RegExp): ValidatorFn {
 export class ForbiddenValidatorDirective implements Validator {
   @Input('appForbiddenName') forbiddenName: string;
 
-  validate(control: AbstractControl): {[key: string]: any} | null {
+  validate(control: AbstractControl): Record<string, any> | null {
```

Comment:
```suggestion
  validate(control: AbstractControl): ValidationErrors | null {
```

And you would need to import `ValidationErrors` from `@angular/forms`.

Reply:
Done, I've cleaned up some unused imports too in that same file, btw I noticed that there are a lot on unused imports in example projects ..